### PR TITLE
dev/core#3001 escape single quotes when rendering tokens in html format

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
port fix from https://github.com/civicrm/civicrm-core/pull/22285

### CODE REVIEW

**Comment by totten** [comment]
Added the corresponding unit-test and update for the release-notes. Merge on pass.

File: tests/phpunit/CRM/Core/TokenSmartyTest.php [hunk 1]
```diff
@@ -132,6 +132,21 @@ public function getDateFormats(): array {
     ];
   }
 
+  public function testTokenDataEscape() {
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Core/TokenSmartyTest.php:135]: Ug - now you've ported it I'm gonna have to fix up the style issues

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Core/TokenSmartyTest.php:135]: (the reason I didn't port originally)

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Core/TokenSmartyTest.php]: Done - looks like the tests had only been running for 20 mins so ...

**Comment by totten** [comment]
jenkins, test this please

**Comment by totten** [comment]
All green.

**Pull request merged**